include/tests/TestResult.h

```diff
@@ -17,12 +17,21 @@ struct TestResult {
       : name(in.stem()), pass(pass), error(error), diff(diff) {}
 
   // Info about result.
-  const fs::path name;
-  const bool pass;
-  const bool error;
-  const std::string diff;
-};
+  fs::path name;
+  bool pass;
+  bool error;
+  std::string diff;
+
+  TestResult clone() { return TestResult(this->name, this->pass, this->error, this->diff); }
```

Comment:
This should be a copy constructor in TestResult, namely `TestResult(const TestResult &other)`.

Reply:
Looking further, I'm not sure this is ever necessary. But if it were, it should be the copy constructor or possibly `operator=(const TestResult &other)`.